hyperdome_client/hyperdome_client.py

```diff
@@ -312,6 +318,10 @@ def counselor_got_guest(guest_key: str):
                         self.session, self.server, self.uid
                     )
                     self.get_messages_task.setAutoDelete(False)
+                    try:
+                        self.get_messages_task.signals.disconnect()
+                    except TypeError:
+                        pass
```

Comment:
suspicious eyes
You should probably add a comment explaining where the type error comes from, if you expect one to be raised

Reply:
I'm still extremely upset that this line exists, but this one line is the fix for the duplicate actions. .disconnect() removes all observers of a signal (which I believe shouldn't exist anyway but whatever) but since they need to be removed right before the new signal is connected, on the first run the .disconnect() fails, and it raises TypeError if there are no connections because PyQT isn't very logical.